src/sagemaker/processing.py

```diff
@@ -1589,41 +1540,161 @@ def run(  # type: ignore[override]
+
+    def _normalize_args(
+        self,
+        job_name=None,
+        arguments=None,
+        inputs=None,
+        outputs=None,
+        code=None,
+        source_dir=None,
+        kms_key=None,
+        dependencies=None,
+        git_config=None,
+    ):
+        """Normalizes the arguments so that they can be passed to the job run
 
-        if job_name is None:
-            job_name = self._generate_current_job_name(job_name)
+        Args:
+            job_name (str): Name of the processing job to be created. If not specified, one
+                is generated, using the base name given to the constructor, if applicable
+                (default: None).
+            arguments (list[str]): A list of string arguments to be passed to a
+                processing job (default: None).
+            inputs (list[:class:`~sagemaker.processing.ProcessingInput`]): Input files for
+                the processing job. These must be provided as
+                :class:`~sagemaker.processing.ProcessingInput` objects (default: None).
+            outputs (list[:class:`~sagemaker.processing.ProcessingOutput`]): Outputs for
+                the processing job. These can be specified as either path strings or
+                :class:`~sagemaker.processing.ProcessingOutput` objects (default: None).
+            code (str): S3 URI or local path to a file with the user script to run. If
+                ``source_dir`` is specified, then ``code`` must be a path relative to the
+                root of ``source_dir``. When ``code`` is already an S3 URI, then
+                ``source_dir``, ``dependencies``, and ``git_config`` are ignored. (default:
+                None).
+            source_dir (str): S3 URI or local path to a folder with any other containing
+                processing source code dependencies aside from the entry point ``code``
+                file. If provided, the whole folder will be bundled and loaded to the
+                SageMaker processing job. If ``source_dir`` is an S3 URI, it must point to
+                a tar.gz file. (default: None).
+            kms_key (str): The ARN of the KMS key that is used to encrypt the
+                user code file (default: None).
+            dependencies (list[str]): A list of path sto directories (absolute or relative)
+                with any additional libraries that should be exported to the container
+                (default: []).
+            git_config (dict[str, str]): Git configurations used for cloning files to
+                retrieve your processing script.
+        """
+        self._current_job_name = self._generate_current_job_name(job_name=job_name)
 
-        estimator = self._upload_payload(
+        inputs_with_code = self._include_code_in_inputs(
+            inputs,
             code,
-            source_dir,
-            dependencies,
-            git_config,
-            job_name,
+            source_dir=source_dir,
```

Comment:
Shouldn't be *necessary* in the sense that the order here matches the function declaration so it'd still work.

I did it for these because:
- They're defined as keyword args and the order didn't seem obvious as a user
- The fact that FrameworkProcessor inserts `source_dir` into the arg order defined by the parent, rather than only appending arguments in the child class, meant that being a bit more explicit could maybe help avoid confusion.

They can go back to positional if needed I'd say